python/cuml/common/base.pyx

```diff
@@ -209,29 +251,53 @@ class Base:
         self.handle = cuml.common.handle.Handle()
 
     def __getattr__(self, attr):
+        """
+        Method gives access to the correct format of cuml Array attribute to
+        the users. Any variable that starts with `_` and is a cuml Array
+        will return as the cuml Array converted to the appropriate format.
+        """
         real_name = '_' + attr
-        if real_name in self.__dict__.keys():
-            if isinstance(self.__dict__[real_name], Array):
+        if hasattr(self, real_name):
+            if isinstance(self.__dict__[real_name], cumlArray):
                 return self.__dict__[real_name].to_output(self.output_type)
 
-    def _get_output_type(self, output_dtype):
-        if isinstance(output_dtype, str):
-            output_type = output_dtype.lower()
-            if output_type in ['numpy', 'cupy', 'cudf', 'numba']:
-                return output_dtype
-        else:
-            self.output_type = self.output_type
-
-    def _check_output_type(self, input):
+    def _set_output_type(self, input):
         """
-        Method to be called by fit/predict/etc. method of inheriting classes
+        Method to be called by fit methods of inheriting classes
         to correctly set the output type depending on the type of inputs,
         class output type and global output type
         """
         if self.output_type == 'input' or self._mirror_input:
-            if type(input) in _input_type_to_str.keys():
-                self.output_type = _input_type_to_str[type(input)]
-            elif is_cuda_array(input):
-                self.output_type = 'numba'
-            else:
-                self.output_type = 'cupy'
+            self.output_type = _input_to_type(input)
+
+    def _get_output_type(self, input):
+        """
+        Method to be called by predict/transform methods of inheriting classes
+        to correctly set the output type depending on the type of inputs,
+        class output type and global output type
+        """
+        if self._mirror_input:
+            return _input_type_to_str[type(input)]
+        else:
+            return self.output_type
+
+
+# Internal, non class owned helper functions
+
+def _input_to_type(input):
+    if type(input) in _input_type_to_str.keys():
```

Comment:
I was kind of confused by what this function does, so I think it could use a comment.